### PR TITLE
[Bugfix] Fixing undeterministic PCC in Kimi block test

### PR DESCRIPTION
**Pull request by ddjekicTT**
### Summary

Kimi prefill block tests used random weights initialized by default in the Torch model. The problem was that the gate bias tensors are by default initialized with torch.empty which not only gave gibberish values causing low PCC but also is not subject to the RNG seed and caused the undeterminizm. 

### Fix

This pr introduces the use of real weights in the Block test and ensures the gate bias is initialized with torch.zeros by default if real weights are not found.

- [(Galaxy) DeepSeek Prefill tests](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-deepseek-prefill-tests.yaml?query=branch:ddjekic/kimi_block_debug)

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:ddjekic/kimi_block_debug)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:ddjekic/kimi_block_debug)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:ddjekic/kimi_block_debug)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:ddjekic/kimi_block_debug)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:ddjekic/kimi_block_debug)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:ddjekic/kimi_block_debug)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:ddjekic/kimi_block_debug)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:ddjekic/kimi_block_debug)

### CODE REVIEW

File: models/demos/deepseek_v3_d_p/tests/test_prefill_block.py [hunk 1]
```diff
@@ -43,12 +44,31 @@
...
     extract_layer_state_dict,
     get_4d_causal_mask,
+    load_and_compute_layer_by_layer,
     tokenize_prompt_to_isl,
 )
 
 _PROMPT_PATHS = {"abc_1k": ABC_1K_PATH, "prompt_5k": PROMPT_5K_PATH, "prompt_25k": PROMPT_25K_PATH}
 from tests.ttnn.utils_for_testing import comp_pcc
 
+@pytest.fixture
```
[the remark this answers is not in the document]
Comment (mbezuljTT) [reply models/demos/deepseek_v3_d_p/tests/test_prefill_block.py:55]: please remove

File: models/demos/deepseek_v3_d_p/tests/test_prefill_block.py [hunk 2]
```diff
@@ -43,12 +44,31 @@
...
+def model_path(variant):
+    """Block-test-scoped override of the shared ``model_path`` fixture.
+    Uses a local pretrained checkpoint if one is already present (env var / default / shared
```
Comment (mbezuljTT) [inline models/demos/deepseek_v3_d_p/tests/test_prefill_block.py:58]: dont we already have this in conftest?

Comment (ddjekicTT) [reply models/demos/deepseek_v3_d_p/tests/test_prefill_block.py:58]: Currently, if the block test doesn't find the model weights it will proceed to download them. However, we don't want this behavior for Kimi 2.6 since we need to upload the unquantized weights manually. I added this fixture override to avoid the test downloading the weights for Kimi and proceed to random weights instead.

This changes the test behavior for Kimi only. I guess i could remove this, but I would have to make less elegant changes elsewehere.

**Review by kgrujcic-tt — APPROVED** [review]

**Review by nbabinTT — APPROVED** [review]
Thank you!

**Review by pavlepopovic — APPROVED** [review]
